### PR TITLE
feat(addon-docs): named colors with ColorPalette

### PR DESCRIPTION
**Pull request by frassinier**
Issue:
Cannot display design token names using ColorPalette docs component

## What I did
Be able to pass an object for colors in addition to a simple array.
image

## How to test

`NamedColors` story has been added.

### CODE REVIEW

**Comment by frassinier** [comment]
👋 @shilman, 
I'm wondering why did you add the 6.0.0 label since there are no breaking changes? 
Because it's a feature request?

**Comment by shilman** [comment]
Yeah. It's not a bugfix, and 6.0 is the next feature release (end of march, with alphas starting later this week)

**Comment by frassinier** [comment]
Looks good to me and what do you think about the changes? :)

Télécharger Outlook pour Android<https://aka.ms/ghei36>

________________________________

File: lib/components/src/blocks/ColorPalette.tsx
Comment (frassinier) [reply lib/components/src/blocks/ColorPalette.tsx]: Thank you for your review @RuneKR but I don't get it?
I simply iterate over the object as it is. 
I mean if design tokens need to be reordered then, it must be done in the story, I guess.

Comment (RuneKR) [reply lib/components/src/blocks/ColorPalette.tsx]: Looks like you got the gist of it. I understand that the order is arbitrary in this solution. I was wondering if the solution could/should be extended to allow the developer to specify the order of the colours. This could be achieved by using this data structure:

```
[{name: 'name of colour', value: 'hex or whatever']
```

This capability can also be added without making it a breaking change.

Comment (frassinier) [reply lib/components/src/blocks/ColorPalette.tsx]: Actually, it's an excellent question! 
I still use design tokens starting by `$` (because of sass variable conventions) so it does not concern me directly.
It would be great to have the point of view of someone else because I consider that it's not totally necessary. :)

Comment (ndelangen) [reply lib/components/src/blocks/ColorPalette.tsx]: I feel like that could be enhancement made in a following PR, if desired.

**Review by ndelangen — APPROVED** [review]

**Review by jsomsanith-tlnd — APPROVED** [review]

**Review by jsomsanith — APPROVED** [review]

**Comment by frassinier** [comment]
Do we need another review, to be able to merge? 🤗

**Comment by shilman** [comment]
cc @domyen

File: lib/components/src/blocks/ColorPalette.tsx
Comment (gregmartyn) [reply lib/components/src/blocks/ColorPalette.tsx]: I wouldn't call the order arbitrary. (Ignoring IE)
https://www.stefanjudis.com/today-i-learned/property-order-is-predictable-in-javascript-objects-since-es2015/

**Comment by ndelangen** [comment]
@frassinier did you delete the branch on purpose or by accident?

**Comment by domyen** [comment]
I didn't really get a chance to check this out in time (sorry everyone). 

**Some changes requests**
The color names should be the same size as the hex colors. Add a 2px margin between the lines. 
image

And [this story's](https://www.chromaticqa.com/snapshot?appId=5a375b97f4b14f0020b0cda3&id=5e1f4bf93a70750020f16554) layout should not have changed because there are no color names included.

**Comment by frassinier** [comment]
I'm sorry guys, I used to delete the branch once it's merged, I did not realize that demo disappears 😬
Dom, the story has changed because I removed an extra div.
Do you need a revert?

**Comment by ndelangen** [comment]
No revert needed, but a follow up PR with @domyen's review comments applied would be great 🙇

**Comment by ndelangen** [comment]
I invited you to the github org @frassinier, you should be able to directly commit on this repo now, no need for a fork.

**Comment by frassinier** [comment]
Thanks, i'm out the office right now, I will do it on next Monday.

**Pull request merged**
